src/plugins/action_trim_responses/src/databaseMethods.ts

```diff
@@ -0,0 +1,77 @@
+const databaseMethods = (DBConnect: any) => ({
+  getAllApplicationResponses: async (applicationId: number) => {
+    const text = `
+    SELECT ar.id, code, value, time_created
+    FROM application_response ar JOIN template_element te
+    ON ar.template_element_id = te.id
+    WHERE application_id = $1
+    ORDER BY time_created
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [applicationId] })
+      const responses = result.rows
+      return responses
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getAllReviewResponses: async (reviewId: number) => {
+    const text = `
+    SELECT rr.id, code, comment, decision, rr.time_created
+    FROM review_response rr JOIN application_response ar
+    ON rr.application_response_id = ar.id
+    JOIN template_element te ON ar.template_element_id = te.id
```

Comment:
Maybe I'm confused -- this action is not about responses that are linked to other reviews, but for duplicate responses for the *same* review. i.e. the ones that get created for a "Changes Requested". So it shouldn't make any difference what level or how many other reviews exist, this just gets all the responses for one specific review (or application), which shouldn't change as it goes back and forth. Is there another case that I'm not seeing?

Reply:
for consolidator, review_response does not link to template_element_id (as far as i know, maybe we should link it on creation)

Reply:
The review_response has an `application_response_id` field, and I assumed that this would be kept unchanged as the reviews went up the chain (so even a top-level review_response would still have a reference to the underlying application_response). In which case, it shouldn't matter how far up the chain the review_response is. Is that not correct?